jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/LambdaBuilder.kt

```diff
@@ -152,6 +171,18 @@ abstract class LambdaBuilder {
             zipLocation.toPath()
         }
 
+    /**
+     * Returns the build directory of the project. Create this if it doesn't exist yet.
+     */
+    private fun getOrCreateBuildDirectory(module: Module): File =
+        WriteAction.computeAndWait<File, Throwable> {
```

Comment:
What is making this a WriteAction?

Reply:
Weird thing is that if I don't have this `WriteAction`, a threading exception (see below) will be thrown in [here](https://github.com/aws/aws-toolkit-jetbrains/blob/master/jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/sam/SamTemplateUtils.kt#L28).

```
java.lang.Throwable: Do not call synchronous refresh under read lock (except from EDT) - this will cause a deadlock if there are any events to fire.
```

Maybe I'm wrong, but it seems when you are creating files under a module content root, the VFS will be rebuilt, therefore a `WriteAction` is needed here.

Reply:
The refresh probably triggers write action in the VFS if it has no entry yet.